spanner/transaction.go

```diff
@@ -533,14 +565,22 @@ func (t *ReadOnlyTransaction) begin(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
+		var md metadata.MD
 		res, err = sh.getClient().BeginTransaction(contextWithOutgoingMetadata(ctx, sh.getMetadata()), &sppb.BeginTransactionRequest{
 			Session: sh.getID(),
 			Options: &sppb.TransactionOptions{
 				Mode: &sppb.TransactionOptions_ReadOnly_{
 					ReadOnly: buildTransactionOptionsReadOnly(t.getTimestampBound(), true),
 				},
 			},
-		})
+		}, gax.WithGRPCOptions(grpc.Header(&md)))
+
+		if GFELatencyMetricsEnabled && md != nil && t.ct != nil {
```

Comment:
Also here: Move the error handling code for the `BeginTransaction` call to before the block that tries to extract the GFE latency.